compiler-rt/lib/hwasan/hwasan_dynamic_shadow.cpp

```diff
@@ -36,15 +36,18 @@ decltype(__hwasan_shadow)* __hwasan_premap_shadow();
 
 namespace __hwasan {
 
+// We cannot call anything in libc here (see comment above), so we need to
+// assume the biggest allowed page size.
+static constexpr uptr kMaxGranularity = 16384;
```

Comment:
Won't it be 64k rather than 16k? Since that is the max page size for aarch64 linux?

Reply:
android max page size is 16k, defined here: https://android.googlesource.com/platform/bionic/+/main/libc/platform/bionic/page.h#41